Neos.Neos/Tests/Functional/Domain/NodeUriTest.php

```diff
@@ -28,6 +38,33 @@ class NodeUriTest extends AbstractNodeTest
      */
     protected $nodeContextPath = '/sites/uri-test/home';
 
+    /**
+     * @var TransientMemoryBackend
+     */
+    protected $transientRouteCacheBackend;
+
+    /**
+     * @var TransientMemoryBackend
+     */
+    protected $transientResolveCacheBackend;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        /** @var RouterCachingService $routerCachingService */
+        $routerCachingService = ObjectAccess::getProperty($this->router, 'routerCachingService', true);
+        $this->transientRouteCacheBackend = new TransientMemoryBackend();
+        $routeCache = new VariableFrontend('Flow_Mvc_Routing_Route', $this->transientRouteCacheBackend);
+        $this->transientRouteCacheBackend->setCache($routeCache);
+        ObjectAccess::setProperty($routerCachingService, 'routeCache', $routeCache, true);
+
+        $this->transientResolveCacheBackend = new TransientMemoryBackend();
+        $resolveCache = new VariableFrontend('Flow_Mvc_Routing_Resolve', $this->transientResolveCacheBackend);
+        $this->transientResolveCacheBackend->setCache($resolveCache);
+        ObjectAccess::setProperty($routerCachingService, 'resolveCache', $resolveCache, true);
```

Comment:
This is a little nasty, but I couldn't think of an easier way to replace the cache backend in order to track the cache tag behavior "end2end"